milli/src/search/new/mod.rs

```diff
@@ -63,7 +63,8 @@ pub struct SearchContext<'ctx> {
     pub phrase_interner: DedupInterner<Phrase>,
     pub term_interner: Interner<QueryTerm>,
     pub phrase_docids: PhraseDocIdsCache,
-    pub restricted_fids: Option<Vec<u16>>,
+    pub restricted_tolerant_fids: Option<Vec<u16>>,
+    pub restricted_exact_fids: Option<Vec<u16>>,
```

Comment:
We now have 2 different restricting lists used allowing us to retrieve the exact words only or the tolerant words only

Reply:
This is a nitpick, but we could consider having a struct here:

```rust
struct RestrictedFids {
    pub tolerant: Vec<FieldId>,
    pub exact: Vec<FieldId>,
}
```

This way, we could have:

```rust
struct SearchContext<'ctx> {
    // ...
    pub restricted_fids: Option<RestrictedFids>,
}
```

This is not about saving a boolean, but rather it would enable two cool things:

1. Make the state where `restricted_exact_fids` is `None`, and `restricted_tolerant_fids` is `Some` irrepresentable. Making impossible states irrepresentable is a good thing, as it spares us from having to handle impossible cases.
2. Allow for methods on the struct, for instance:
```rust
impl RestrictedFieldIds {
    pub fn contains(&self, fid: &FieldId) -> bool {
        self.tolerant.contains(fid) || self.exact.contains(fid)
    }
}
```
(I recommend not over-doing it, and only adding the methods that you actually use)

then the code in DB cache that does:

```rust
        if self.restricted_fids.as_ref().map_or(false, |fids| !fids.contains(&fid)) {
```

does not need to change at all 😎 

again, nitpick, feel free to tell you don't want to do it for now (or at all).

Reply:
Done @dureuill :D